src/shared/file-record.js

```diff
@@ -130,12 +130,22 @@ class FileSystemHost {
       const subpath = splitPath.join(this.path.sep);
       const rawRecord = await this.readFS(this.path.join(root, subpath));
       if (isDirectory(rawRecord)) {
+        const entries = [];
+        for (const { name } of rawRecord.entries) {
+          const path = this.path.join(subpath, name);
+          try {
+            const stat = await this.fs.stat(this.path.join(root, path));
+            if (matchesGlob(path, stat.isDirectory())) {
+              entries.push({ name });
+            }
+          } catch (error) {
+            console.error(`Error reading filesystem ${this.path.join(root, path)}: ${error}`);
+          }
```

Comment:
I appreciate the care you're taking here, but this seems like a failure which could impact the validity of the results. If it occurred in the GitHub Actions environment, logging a warning would only help us verify that something went wrong after reporting faulty results--it wouldn't stop them from being reported at all. What do you say about crashing for this case?

Reply:
Cool